libs/core-ui/src/lib/util/datasetUtils/getColumnRanges.ts

```diff
@@ -0,0 +1,203 @@
+    );
+  }
+  return ranges;
+}
+
+function getIndexFeatureRange(dataset: IDataset): INumericRange {
+  return {
+    max: dataset.features.length - 1,
+    min: 0,
+    rangeType: RangeTypes.Integer
+  };
+}
+
+function getDatasetFeatureRange(
+  dataset: IDataset,
+  column: string
+): INumericRange | ICategoricalRange {
+  const featureIndex = dataset.feature_names.findIndex(
+    (item) => item === column
+  );
+  const featureVector = dataset.features.map((row) => row[featureIndex]);
+  const isCategorical = dataset.categorical_features.includes(column);
+  return isCategorical
+    ? ({
+        rangeType: RangeTypes.Categorical,
+        uniqueValues: _.uniq(featureVector).sort()
+      } as ICategoricalRange)
+    : ({
+        max: _.max(featureVector) || 0,
+        min: _.min(featureVector) || 0,
+        rangeType: featureVector.every((val) => Number.isInteger(val))
+          ? RangeTypes.Integer
+          : RangeTypes.Numeric
+      } as INumericRange);
+}
+
+function getClassificationErrorFeatureRange(
+  modelType: ModelTypes
+): ICategoricalRange | undefined {
+  if (IsBinary(modelType)) {
+    return {
+      rangeType: RangeTypes.Categorical,
+      uniqueValues: [
+        localization.Interpret.Columns.trueNegative,
+        localization.Interpret.Columns.falsePositive,
+        localization.Interpret.Columns.falseNegative,
+        localization.Interpret.Columns.truePositive
+      ]
+    };
+  }
+  if (IsMulticlass(modelType)) {
+    return {
+      rangeType: RangeTypes.Categorical,
+      uniqueValues: [
+        localization.Interpret.Columns.correctlyClassified,
+        localization.Interpret.Columns.misclassified
+      ]
+    };
+  }
+  return;
+}
+
+function getRegressionErrorFeatureRange(
+  dataset: IDataset,
+  modelType: ModelTypes
+): INumericRange | undefined {
+  if (modelType === ModelTypes.Regression && dataset.predicted_y) {
+    const regressionErrors = [];
+    for (let index = 0; index < dataset.features.length; index++) {
+      const trueY = dataset.true_y[index];
+      const predictedY = dataset.predicted_y[index];
+      if (Array.isArray(trueY) || Array.isArray(predictedY)) {
+        return;
+      }
+      regressionErrors.push(Math.abs(trueY - predictedY));
+    }
+    return {
+      max: _.max(regressionErrors) || 0,
+      min: _.min(regressionErrors) || 0,
+      rangeType: RangeTypes.Numeric
+    };
+  }
+  return;
+}
+
+function getRange(
+  dataset: IDataset,
+  modelType: ModelTypes,
+  values: number[] | number[][],
+  property: string,
+  ranges: { [key: string]: INumericRange | ICategoricalRange }
+): void {
+  let categoricalValues = dataset.class_names;
+  // if it is 1D array
+  if (!Array.isArray(values[0])) {
+    if (modelType === ModelTypes.Regression) {
+      const numbers: number[] = [];
+      // this for loop is only to let it make sure values is a 1D array, so it can be used with _.max and _.min
+      values.forEach((value) => {
+        if (!Array.isArray(value)) {
```

Comment:
do we want to raise error if some value here is an array?

Reply:
or is there any existing check somewhere else already?

Reply:
it is handled in the below else statement - from line 181 to the end